### PR TITLE
feat(auth): implement the sign in screen and tests

### PR DESCRIPTION
**Pull request by rBergCode**
# Summary
The objective was to have a Sign In screen with accordance to the [Figma](https://www.figma.com/design/EOM0DPxmM1FzGwxFeoItsE/HikeMate-wireframe-design?node-id=184-2&node-type=frame&t=SsRYKh3K4diq0rUb-0) design.
For this I created the `SignInScreen` composable and made it look like the design.
I also added a `SignInScreenTest` to test the screen

Finally, I added multiple assets as:
- The background image of the sign in screen
- The google logo for the button
- The app icon
- The selected font

### CODE REVIEW

File: app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import androidx.test.espresso.intent.Intents
+import ch.hikemate.app.MainActivity
+import com.kaspersky.kaspresso.testcases.api.testcase.TestCase
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class SignInScreenTest : TestCase() {
+  @get:Rule val composeTestRule = createAndroidComposeRule<MainActivity>()
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt:17]: Question: (Just for my understanding), what is `@get:Rule`?

File: app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  // The IntentsTestRule is not reliable.
```
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt:19]: Nitpick: Is this comment still relevant? What does it apply to? What is his purpose?

File: app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  Scaffold(
+      modifier = Modifier.fillMaxSize(),
+      content = { padding ->
+        Box(
+            modifier = Modifier.fillMaxSize(),
+        ) {
+          // I created a box to make the background image take a little more space
+          // than the screen size, so that the blur effect doesn't show white edges
+          Box(
+              modifier = Modifier.fillMaxSize().clipToBounds().scale(1.1f),
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt:51]: Thanks for the explanation!

File: app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+                      TextStyle(
+                          color = Color.White,
+                          fontFamily = FontFamily(Font(R.font.kaushan_script, FontWeight.Bold)),
+                          fontSize = 60.sp,
+                          fontWeight = FontWeight.Bold,
+                      ),
+              )
+            }
+
+            SignInWithGoogleButton { /* Handle sign in click here */}
```
Comment (Wylwi) [inline app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt:86]: Don't forget to handle the sign in later on and to test it later!

File: app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt (hunk 2 above)
Comment (Wylwi) [inline app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt:19]: Nitpick: Is this comment still relevant? What does it apply to? What is his purpose?

**Review by Wylwi — DISMISSED** [review]
Only small modifications/questions, otherwise, pretty perfect code!

File: app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt (hunk 1 above)
Comment (rBergCode) [reply app/src/androidTest/java/ch/hikemate/app/ui/auth/SignInScreenTest.kt:17]: `@get:Rule` annotation indicates that the property composeTestRule is a JUnit test rule

**Review by Wylwi — APPROVED** [review]
LGTM now!

**Pull request merged**